tests/unit/gtsysfs.cpp

```diff
@@ -81,4 +80,158 @@ TEST(LibopaecsysfsCommonMOCKHW, fpga_sysfs_01) {
 	result = sysfs_deviceid_from_path("/sys/class/fpga/intel-fpga-dev/intel-fpga-fme", &deviceid);
 	ASSERT_NE(result, FPGA_OK);
 
+}
+
+/**
+* @test    fpga_sysfs_02
+* @brief   Tests: sysfs_read_int,sysfs_read_u32
+*          sysfs_read_u32_pair,sysfs_read_u64
+*          sysfs_read_u64,sysfs_write_u64
+*..........get_port_sysfs,sysfs_read_guid
+*..........get_fpga_deviceid
+*/
+TEST(LibopaecsysfsCommonMOCKHW, fpga_sysfs_02) {
+
+	fpga_result result;
+	struct _fpga_token _tok;
+	fpga_token tok = &_tok;
+	fpga_handle h;
+	int i;
+	uint32_t u32;
+	uint32_t u1;
+	uint32_t u2;
+	uint64_t u64;
+
+
+	//Invalid input path
+	result = sysfs_read_int("", NULL);
+	EXPECT_NE(result, FPGA_OK);
+
+	//NULL input parameters
+	result = sysfs_read_int(NULL, NULL);
+	EXPECT_NE(result, FPGA_OK);
+
+	//Invalid input path
+	result = sysfs_read_int("/tmp/class/fpga/intel-fpga-dev.0/intel-fpga-fme.10", NULL);
+	EXPECT_NE(result, FPGA_OK);
+
+	result = sysfs_read_int("/tmp/class/fpga/intel-fpga-dev.0/intel-fpga-fme.0", NULL);
+	EXPECT_NE(result, FPGA_OK);
+
+	// Valid input path
+	result = sysfs_read_int("/tmp/class/fpga/intel-fpga-dev.0/intel-fpga-fme.0/socket_id", &i);

```

Comment:
How is this a valid path? It's only valid when using with the mock driver. If this is run without the mock driver, on real hw, then this will fail.
Out mock implementation rewrites /sys to /tmp to technically you can put any three letter starting directory, however, that will ONLY work with mock. It would be better to assume real HW and let mock do the rewrite.